server/lib/ReportManager.js

```diff
@@ -236,4 +235,12 @@ class ReportManager {
   }
 }
 
+// function to encode file data to base64 encoded string
+function base64Encode(file) {
+  // read binary data
+  const bitmap = fs.readFileSync(file);
```

Comment:
This might be part of the problem with the speed - we don't want to write synchronous code on the server!  Can you make this into an asynchronous operation?